crates/rust-client/src/store/web_store/ts/transactions.ts

```
@@ -127,9 +127,13 @@ export async function getTransactions(filter: string) {
 }
 
 export async function insertTransactionScript(
-  scriptRoot: Uint8Array,
+  scriptRoot?: Uint8Array,
   txScript?: Uint8Array
 ) {
+  if (!scriptRoot) {
+    return;
+  }
+
```

Maybe we should throw an exception here, the case of inserting a transactionScript without root seems weird 🤔

---

Transactions could be instantiated without a transaction script, so in that case we don't want to insert it into the store. I'll refactor the sync process so that this function is only called when the root is not null.